DeploymentLogicTest.cpp

```diff
@@ -0,0 +1,404 @@
+    float magy; // 4 bytes - 56
+    float magz; // 4 bytes - 60
+    float altitude; // 4 bytes - 64
+    float temp; // 4 bytes - 68
+    //float pressure;
+    float w; // 4 bytes - 72 (quaternion nonsense)
+    float x; // 4 bytes - 76
+    float y; // 4 bytes - 80
+    float z; // 4 bytes - 84
+    long long checksum; // >:C   8 bytes - 92
+} Packet; //22 values
+
+
+
+//function definitions:
+Packet extract_values(std::string values);
+std::vector<Packet> packet_maker(int* num_packets, std::vector<int>* line_breaks); 
+// ^ look at address of numpackets - modifies oG variable it pointing to 
+//line breaks is the position of line breaks
+
+float accessPacketFloat(Packet packet, int index);
+//vector refers to a library 
+
+//************************************************************* MAIN  *********************************
+int main() 
+{
+
+
+//vector of packets (basically an array but not really)
+    std::vector<Packet> packet_vector;
+    std::vector<int> line_breaks;
+    
+    //ifstream copyData;
+    std::ofstream outData;
+    std::string values = "";
+    
+    int num_packets = 0;
+
+    packet_vector = packet_maker(&num_packets, &line_breaks);
+
+#define abcd packet_vector[num_packets] //abcd = packet_vectors[num_packets]
+    std::cout << "num_packets: " << num_packets + 1 << "  last quaternion: " << abcd.w << " " << abcd.x << " " << abcd.y << " " << abcd.z << " " << std::endl;
+    std::cout << "some checksums: " << abcd.checksum << " " << packet_vector[num_packets - 1].checksum << " " << packet_vector[num_packets - 2].checksum << " " << std::endl;
+
+    //setup done, time for the rocket simulation
+
+
+    constexpr int packet_mem_size = 60; //literally a constant. why are we so dramatic 'constexpr?
+    //packet memory size  = packet_mem_size
+    //there are 60 packets. everytime a new apcket is introduced the other is kicked out
+
+    std::array<Packet, packet_mem_size> packet_memory;
+    Packet rolling_integral = packet_vector[0]; //rolling integral - changes as time passes (not in use rn)
+    Packet rolling_derivative = packet_vector[0]; // 0 means we getting intiaitve val
+    std::array<float, packet_mem_size> deriv_altitude;
+    std::array<float, packet_mem_size> deriv_accz;
+
+    long delta_time  = 0;
+
+    //sets up the packet memory with first packet_mem_size packets (initializes the packets)
+    packet_memory[0] = packet_vector[0];
+    deriv_altitude[0] = 0;
+    deriv_accz[0]=0;
+    for (int i = 1; i < packet_mem_size; i++) 
+    {
+        packet_memory[i] = packet_vector[i];
+        delta_time = (packet_memory[i].time - packet_memory[i-1].time)/1000000;
+
+       // deriv_pressure[i] = (packet_memory[i].pressure - packet_memory[i - 1].pressure) * ((packet_memory[i].delta_time)/1000000);
+        deriv_altitude[i] = (packet_memory[i].altitude - packet_memory[i - 1].altitude) * (delta_time);
+        deriv_accz[i] = (packet_memory[i].accz - packet_memory[i - 1].accz) * (delta_time);

```

Comment:
So I primarily used the keith lawndart data set and and looked at the changes in acceleration to see if there was any pattern. For example, at apogee, acceleration fluctuated slightly and stayed within the ranges of -9.7 to -9.8 (which is consistent with the physics I've learned to date). I used the change in acceleration because at certain points when the acceleration jumped or dropped, it showed a change in stage (ex: when the engine first fired there was a massive change in acceleration). 

Plus, it lets me be more flexible in my ranges. By having change in acceleration instead of just pure acceleration, the staging will still apply to a variety of launches. 

I can still also use velocity because I did notice patterns in the velocity as well but I wanted your feedback on what was already laid down.